lua/rocks/operations.lua

```diff
@@ -482,20 +482,22 @@ operations.update = function()
                 name = name,
                 version = rock.target_version,
             })
-            local success, ret = pcall(future.wait)
+            local co = coroutine.create(future.wait)
+            local success, ret = coroutine.resume(co)
             ct = ct + 1
             nio.scheduler()
-            if not success then
+            if success then
+                user_rocks.plugins[ret.name] = ret.version
+                progress_handle:report({
+                    message = ("Updated %s: %s -> %s"):format(rock.name, rock.version, rock.target_version),
+                    percentage = get_percentage(ct, #outdated_rocks),
+                })
+            else
                 report_error(("Failed to update %s."):format(rock.name))
                 progress_handle:report({
                     percentage = get_percentage(ct, #outdated_rocks),
                 })
             end
-            user_rocks.plugins[ret.name] = ret.version
-            progress_handle:report({
-                message = ("Updated %s: %s -> %s"):format(rock.name, rock.version, rock.target_version),
-                percentage = get_percentage(ct, #outdated_rocks),
-            })
         end
```

Comment:
This was a bug I fixed on the fly - we shouldn't be attempting to index `[ret.name]` if the function that produces `ret` failed.